### ISSUE TITLE
P2 audit: distinguish nominal from realized M4 travel-condition loss per migration

### ISSUE THREAD

**Issue by Datengu** [Datengu/anthrosim#225]
## Scientific observability finding

M4 records `travelConditionCostPerPerson` on every `HouseholdMigration` event / decision trace, but that value is the **nominal configured decrement** derived from distance:

```text
min(travel_condition_cost_per_cell × distance_cells, 1000)
```

The actual condition update is saturating for each mover:

```rust
after = before.saturating_sub(condition_cost)
realized_loss = before - after
```

Therefore the event's nominal cost can exceed the condition actually lost by one or more movers.

Example:

```text
nominal travelConditionCostPerPerson = 100
mover A condition before = 1000 -> realized loss 100
mover B condition before = 50   -> realized loss 50
```

The event/trace reports `100` for the household, while realized household loss is `150`, not `200`.

`MigrationSummary.travelConditionCostTotal` correctly accumulates realized loss from `Population::apply_household_relocations`, but the per-move event does not preserve the realized amount, and current resource/condition observability does not preserve historical per-person condition before each move.

## Why this matters

A preserved run can explain which migration occurred and what nominal penalty was requested, but cannot reliably attribute the **realized condition change** to an individual move from the event history alone.

This matters for causal diagnosis and sensitivity because migration-induced condition loss feeds subsequent model state and, under current open issue #200, can also affect mortality labelled as resource scarcity. A per-move analysis that multiplies `peopleMoved × travelConditionCostPerPerson` will overstate realized loss whenever movers are near the lower condition bound.

The migration research document correctly describes the manifest's aggregate as `total realized travel-condition loss`, but the detailed trace simply calls its field `travel condition cost per person`, leaving nominal versus realized semantics easy to conflate.

## Required work

- Make the event/trace field explicitly nominal/applied, e.g. `nominalTravelConditionCostPerPerson`, if retained.
- Preserve the realized condition loss for each completed household move, at minimum as `realizedTravelConditionLossTotal` on the migration event/trace.
- If scientifically useful, additionally preserve mover count affected by saturation or pre/post household condition summaries without requiring full per-person logging.
- Reconcile the sum of per-move realized losses exactly to `MigrationSummary.travelConditionCostTotal`.
- Update documentation and Explorer labels to distinguish:
  - configured/nominal decrement;
  - realized bounded decrement.
- Add tests where no mover saturates, some movers saturate, and all movers saturate at zero.
- Coordinate with #215 so the minimum condition-history observability surface is coherent rather than duplicated.

## Acceptance

A preserved migration event history can reproduce the aggregate realized M4 travel-condition loss exactly, and consumers cannot mistake the nominal per-person decrement for the condition actually lost.

**Research gate:** required before per-move migration physiological/condition effects are quantitatively interpreted; not a blocker for analyses using only migration counts/destinations.
